vcr/credential/util.go

```diff
@@ -160,22 +159,10 @@ outer:
 				continue
 			}
 		}
-		bl := make([]BaseCredentialSubject, 0)
-		err = credential.UnmarshalCredentialSubject(&bl)
-		if err != nil {
+		subjectDID, err := credential.SubjectDID()
+		if err == nil && !slices.Contains(didMethods, subjectDID.Method) {
 			continue
 		}
-		for _, b := range bl {
-			if b.ID != "" {
-				// check credentialSubject
-				subjectDID, err := did.ParseDID(b.ID)
-				if err == nil {
-					if !slices.Contains(didMethods, subjectDID.Method) {
-						continue outer
-					}
-				}
-			}
-		}
```

Comment:
this used to require only one subjectDID to be valid, but `credential.SubjectDID()` now requires all to be valid and have the same value.